hyper/http20/connection.py

```diff
@@ -254,7 +255,10 @@ def connect(self):
                 proto = H2C_PROTOCOL
 
             log.debug("Selected NPN protocol: %s", proto)
-            assert proto in H2_NPN_PROTOCOLS or proto == H2C_PROTOCOL
+            # AppEngine SSLSocket does not have selected_npn_protocol nor
+            # selected_alpn_protocol, so just avoid this assertion.
+            assert is_appengine or (
+                proto in H2_NPN_PROTOCOLS or proto == H2C_PROTOCOL)
```

Comment:
Sorry, I think I wasn't clear enough with my previous feedback. When I was asking about `H2C_PROTOCOL` I wanted to know why you chose that over `h2`. `h2c` is meant to be reserved for HTTP/2 via opportunistic encryption, which seemed an odd choice. I'd be happy to use `h2`.